test/Utils.hs

```diff
@@ -66,13 +66,16 @@ printProgramResult = printProcResult . fromProgramResult
 type AgdaArgs = [String]
 
 
-readAgdaProcessWithExitCode :: AgdaArgs -> Text
+readAgdaProcessWithExitCode :: Maybe EnvVars  -- ^ Extra environment variables, unexpanded.
+                            -> AgdaArgs -> Text
                             -> IO (ExitCode, Text, Text)
-readAgdaProcessWithExitCode args inp = do
-  agdaBin <- getAgdaBin
-  envArgs <- getEnvAgdaArgs
+readAgdaProcessWithExitCode extraEnv args inp = do
+  origEnv <- getEnvironment
+  home <- getHomeDirectory
+  let env = expandEnvVarTelescope home $ maybe origEnv (origEnv ++) extraEnv
+  let envArgs = maybe [] words $ lookup "AGDA_ARGS" env
   -- hPutStrLn stderr $ unwords $ agdaBin : envArgs ++ args
-  let agdaProc = (proc agdaBin (envArgs ++ args)) { create_group = True }
+  let agdaProc = (proc (getAgdaBin env) (envArgs ++ args)) { create_group = True , env = Just env }
```

Comment:
@lawcho: Could this change somehow have triggered #6905?

Reply:
So this is likely a bug in the `process` library that only surfaces on `macOS`.  The fix of this bug suggests that setting `cwd` here would workaround this bug, see https://github.com/haskell/process/pull/296/files.

Reply:
My experiments confirm that indeed this PR surfaced the bug in `process` currently affecting GHCs 9.2.8, 9.4.7, and 9.6.3.  Sorry, you stepped on a mine here, you take no blame.